modules/apt.sh

```diff
@@ -43,3 +51,38 @@ install_package_if_missing_file() {
         check_result apt_get install "$package"
     fi
 }
+
+_apt_write_list_file() {
+    local repo_file="$1"
+    local repo_url="$2"
+
+    cat >"$repo_file" <<EOF
+deb https://${repo_url}/ubuntu ${SERIES} main
+# deb-src https://${repo_url}/ubuntu ${SERIES} main
+EOF
+}
+
+_apt_add_auth() {
+    local repo_url="$1"
+    local credentials="$2"
+
+    local login password
+    login=$(echo "$credentials" | cut -d: -f1)
+    password=$(echo "$credentials" | cut -d: -f2)
+    [ -f "$APT_AUTH_FILE" ] || touch "$APT_AUTH_FILE"
+    chmod 600 "$APT_AUTH_FILE"
+    echo "machine ${repo_url}/ubuntu/ login ${login} password ${password}" \
+         >>"$APT_AUTH_FILE"
+}
+
+_apt_remove_auth() {
+    local repo_url="$1"
+
+    local tempfile
+    tempfile=$(mktemp)
+    chmod 600 "$tempfile"
+    awk -v url="$repo_url" \
+        'BEGIN { pattern = "^machine " url "/ubuntu/ "; }; $0 !~ pattern;' \
+        "$APT_AUTH_FILE"  >"$tempfile"
```

Comment:
awk showoff :)
How about:
sed -i "/^machine $url\/ubuntu\/ login/d" $APT_AUTH_FILE

No need to use temp files.

That being said, I think I will file a bug about the suffix "url" being wrong for some of these variables. We are using $foo_url in many places where it's actually a hostname, not an url.

Reply:
```diff
diff --git a/modules/apt.sh b/modules/apt.sh
index 742d471..955027c 100644
--- a/modules/apt.sh
+++ b/modules/apt.sh
@@ -76,13 +76,7 @@ _apt_add_auth() {
 }
 
 _apt_remove_auth() {
-    local repo_url="$1"
+    local repo_hostname="$1"
 
-    local tempfile
-    tempfile=$(mktemp)
-    chmod 600 "$tempfile"
-    awk -v url="$repo_url" \
-        'BEGIN { pattern = "^machine " url "/ubuntu/ "; }; $0 !~ pattern;' \
-        "$APT_AUTH_FILE"  >"$tempfile"
-    mv "$tempfile" "$APT_AUTH_FILE"
+    sed -i "/^machine $repo_hostname\/ubuntu\/ login" "$APT_AUTH_FILE"
 }
```

Reply:
done, thanks